config/src/config/network_config.rs

```diff
@@ -328,15 +328,61 @@ impl NetworkConfig {
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
 #[serde(default, deny_unknown_fields)]
 pub struct PeerMonitoringServiceConfig {
+    pub enable_peer_monitoring_client: bool, // Whether or not to spawn the monitoring client
+    pub latency_monitoring: LatencyMonitoringConfig,
     pub max_concurrent_requests: u64, // Max num of concurrent server tasks
     pub max_network_channel_size: u64, // Max num of pending network messages
+    pub metadata_update_interval_ms: u64, // The interval (ms) between metadata updates
+    pub network_monitoring: NetworkMonitoringConfig,
+    pub peer_monitor_loop_interval_ms: u64, // The interval (ms) between peer monitor loop executions
 }
 
 impl Default for PeerMonitoringServiceConfig {
     fn default() -> Self {
         Self {
+            enable_peer_monitoring_client: false,
+            latency_monitoring: LatencyMonitoringConfig::default(),
             max_concurrent_requests: 1000,
             max_network_channel_size: 1000,
+            metadata_update_interval_ms: 5000,
+            network_monitoring: NetworkMonitoringConfig::default(),
+            peer_monitor_loop_interval_ms: 1000,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct LatencyMonitoringConfig {
+    pub latency_ping_interval_ms: u64, // The interval (ms) between latency pings for each peer
+    pub latency_ping_timeout_ms: u64,  // The timeout (ms) for each latency ping
+    pub max_latency_ping_failures: u64, // Max ping failures before the peer connection fails
+    pub max_num_latency_pings_to_retain: usize, // The max latency pings to retain per peer
+}
+
+impl Default for LatencyMonitoringConfig {
+    fn default() -> Self {
+        Self {
+            latency_ping_interval_ms: 30_000, // 30 seconds
+            latency_ping_timeout_ms: 20_000,  // 20 seconds
```

Comment:
Unfortunately, yes. The default timeout in state sync is 10 seconds (and we still see timeouts in production, which is why we require exponential back-off for large data chunks). I've chosen 20 seconds here because it's the same as the network health checker and we want to provide additional room for delays, otherwise network connections will be more likely to be face false disconnects when the system is under load.

Reply:
Orthogonal to this PR, but we should figure out what is causing 20 seconds delay in network pings, where an average network latency should be less than 500 ms.

Reply:
> Orthogonal to this PR, but we should figure out what is causing 20 seconds delay in network pings, where an average network latency should be less than 500 ms.

Yeah, makes sense. My hope is that it's something simple, e.g., a bottleneck in message sending/receiving in the stack.

Reply:
This message is small, so 10 seconds would be really bad

Reply:
> This message is small, so 10 seconds would be really bad

Annoyingly, this message is pretty much the same size as the existing health checker message (if not exactly the same), and under heavy load we already see those fail today (with the same timeout of 20 seconds). So, I think the issue is more related to queuing and/or blocking of messages, even if they're small. It still requires some digging -- hopefully I can look into at soon 😄